streaming/base/dataset.py

```diff
@@ -265,7 +266,7 @@ def __init__(self,
                                                   size=len(self.shard_sizes) * np.uint8(0).nbytes)
 
         # Destroy process group, and de-initialize the distributed package
-        if is_dist_pg_initialized:
+        if is_dist_pg_initialized and self._rank_world.num_ranks > 1:
```

Comment:
I think this second check isn’t necessary?

Reply:
Fixed in https://github.com/mosaicml/streaming/pull/153/commits/5f922b927a1686dc040d065c09d7770bd1a8640b .